File: 数学/LC1两数之和.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        //由于不存在重复数字，直接用hashMap接口存储，
        //可以直接用get获得下标，其次是通过containsKey来找到key
        Map<Integer, Integer> hashMap = new HashMap<>();
        for(int j=0;;j++){
            int x = hashMap[j];
            if (hashMap.containsKey(target - x)) { // 找到了
                return new int[]{hashMap.get(target - x), j}; // 返回两个数的下标
            }
        }
    }
}

File: 数学/LC1两数之和.cpp
#include<bits/stdc++.h>
#include<vector>
#include<unordered_map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int>temp; // 创建一个空哈希表
        for (int j = 0; ; j++) { // 枚举 j
            // 在左边找 nums[i]，满足 nums[i]+nums[j]=target
            auto it = temp.find(target - nums[j]);
            if (it != temp.end()) { // 找到了
                return {it->second, j}; // 返回两个数的下标
            }
            temp[nums[j]] = j; // 保存 nums[j] 和 j
        }
    }
};